requests/models.py

```diff
@@ -932,6 +944,8 @@ def raise_for_status(self):
             http_error_msg = u'%s Server Error: %s for url: %s' % (self.status_code, reason, self.url)
 
         if http_error_msg:
+            if isinstance(body_text, basestring):
+                http_error_msg += u' Response Body: %s' % body_text
```

Comment:
I agree we can definitely only take a snippet. But recall this would only be triggered IF we were dealing with a 4XX or 5XX status which shouldn't have an enormous body. Limiting the size to a few hundred characters seems like a reasonable thing for those bad cases.